### ISSUE TITLE
cv_gamlr with observational weights

### ISSUE THREAD

**Issue by walterwzhang** [TaddyLab/gamlr#14]
Hi, 

An error occurs when I run the `cv.gamlr` function with observational weights. For example,

```r
require(gamlr)
n <- 1000
p <- 3
xvar <- matrix(0.9, nrow=p,ncol=p)
diag(xvar) <- 1
x <- matrix(rnorm(p*n), nrow=n)%*%chol(xvar)
y <- 4 + 3*x[,1] + -1*x[,2] + rnorm(n)
obs_weights   <- runif(n, 0, 1)
cvfit <- cv.gamlr(x, y, gamma=1, verb=TRUE, obsweight = obs_weights)
```
yields `fold Error: length(obsweight) == n is not TRUE`. 

It seems that the cv.glmnet code does not properly subset the vector of weights when estimating `gamlr` for each fold. I think I have managed to patch the code ([see here](https://gist.github.com/walterwzhang/607d26990d2e3c8c9d686e8b3c3996cc)) by subsetting the `obsweight` vector before each `gamlr` run in the same manner that the `shift` argument has been subsetted in the code.

Thank you for your help and for making the package available.

**Comment by mataddy** [comment]
thanks @walterwzhang I'll look into this.  If you want to do a fork and pull req for your fix that could speed things up.  Thanks!  MT

**Comment by mataddy** [comment]
merged #15.  thanks @walterwzhang !
